subprojects/frameworklauncher/yarn/src/main/java/com/microsoft/frameworklauncher/applicationmaster/StatusManager.java

```diff
@@ -795,6 +795,12 @@ public synchronized int getOutstandingTaskAppearedRound() {
     return outstandingTaskAppearedRound;
   }
 
+  public synchronized ExecutionType getFrameworkExecutionType() throws Exception {
+    final FrameworkRequest frameworkRequest =
+        zkStore.getFrameworkRequest(conf.getFrameworkName());
+    return frameworkRequest.getFrameworkDescriptor().getExecutionType();
+  }
```

Comment:
Pls do not directly get ExecutionType from ZK, reuse the in-memory one.

Reply:
Please, suggest better place for this function. Actually, in AM at least 3 services do similar calls, as I've noticed. BTW, will the in-memory entry be actual?